tests/helpers/fd_utils.cc

```diff
@@ -78,8 +81,31 @@ namespace Pistache
 
         return std::distance(directory_iterator(fds_dir),
                              directory_iterator {});
+#else // fallback case, e.g. *BSD
+#ifndef OPEN_MAX
+#define OPEN_MAX	4096
+#endif
+        long maxfd;
+
+	maxfd = sysconf(_SC_OPEN_MAX);
+        if (maxfd < 0) // or if sysconf not defined at all
+            maxfd = getdtablesize();
```

Comment:
This function has been removed from POSIX; from <https://pubs.opengroup.org/onlinepubs/007908775/xsh/getdtablesize.html>:

> The [getrlimit()](https://pubs.opengroup.org/onlinepubs/007908775/xsh/getrlimit.html) function returns a value of type rlim_t. This interface, returning an int, may have problems representing appropriate values in the future. Applications should use the [getrlimit()](https://pubs.opengroup.org/onlinepubs/007908775/xsh/getrlimit.html) function instead.

Reply:
Well spotted. I swapped in getrlimit(RLIMIT_NOFILE) instead.